ably/personal_task.go

```diff
@@ -137,6 +137,21 @@ func personalTask(testConfig TestConfig) {
 	}
 }
 
+// publishWithRetries makes multiple attempts to publish a message on the given
+// channel.
+func publishWithRetries(channel *ably.RealtimeChannel, name string, data interface{}) (err error) {
+	timeout := 30 * time.Second
+	delay := 100 * time.Millisecond
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(delay) {
+		_, err = channel.Publish(name, data)
+		if err == nil {
```

Comment:
Again, can we not make a distinction between retriable and non-retriable errors? If the connection is validly in a `failed` state, this is just going to loop.

Reply:
ably-go doesn't seem to expose a way to detect this via the API, so I've just skipped the retry for the error we've seen (i.e. `attempted to attach channel to an inactive connection`) in bd69074, and also added a TODO to remove the logic as it really should be handled by ably-go.